src/csharp/Grpc.Tools.Tests/MsBuildEndToEndTest.cs

```diff
@@ -0,0 +1,190 @@
+#region Copyright notice and license
+
+// Copyright 2020 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.IO;
+using NUnit.Framework;
+using System.Diagnostics;
+using System.Reflection;
+using System.Collections.Specialized;
+using System.Collections;
+
+namespace Grpc.Tools.Tests
+{
+    public class MsBuildEndToEndTest
```

Comment:
Nit, as discussed, this isn't really a full End2End test since we're invoking fake protoc.
maybe MsBuildIntegrationTest is a better name?